src/main/java/com/keniu/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.keniu.security;
+
+import com.keniu.exceptions.RegistrationException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final int BEARER_PREFIX_LENGTH = 7;
+    private final JwtUtil jwtUtil;
+    private final UserDetailsService userDetailsService;
+
+    @Override
+    protected void doFilterInternal(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            FilterChain filterChain)
+            throws ServletException, IOException {
+        String token = getToken(request);
+        if (jwtUtil.isValidToken(token)) {
+            UserDetails userDetails = userDetailsService.loadUserByUsername(
+                    jwtUtil.getUsername(token)
+            );
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails,
+                    null, // Password isn't needed, JWT already validated
```

Comment:
let's remove comment

Reply:
I'd leave it as is, since null doesn't carry any meaningful context. Here, it's unclear what we are passing and why this "something" is null